intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/indexing/indexing.py

```diff
@@ -115,110 +140,99 @@ def KB_construct(self, input):
+        elif isinstance(input, List):
+            try:
+                data_collection = self.parse_html(input)
+            except:
+                print("The given link/str is unavailable. Please try another one!")
         else:
-            if os.path.exists(input):
-                if os.path.isfile(input):
-                    data_collection = self.parse_document(input)
-                elif os.path.isdir(input):
-                    data_collection = self.batch_parse_document(input)
-                else:
-                    print("Please check your upload file and try again!")
-                if self.document_store == "inmemory":
-                    document_store = InMemoryDocumentStore(use_gpu=False, use_bm25=True)
-                elif self.document_store == "Elasticsearch":
-                    document_store = ElasticsearchDocumentStore(host="localhost", index=self.index_name,
-                                                                port=9200, search_fields=["content", "title"])
+            print("The input path is invalid!")
 
-                documents = []
-                for data, meta in data_collection:
-                    metadata = {"source": meta}
-                    if len(data) < 5:
-                        continue
-                    new_doc = SDocument(content=data, meta=metadata)
-                    documents.append(new_doc)
-                assert documents != [], "The given file/files cannot be loaded."
-                document_store.write_documents(documents)
-                print("The local knowledge base has been successfully built!")
-                return document_store
-            else:
-                print("There might be some errors, please wait and try again!")
+        if self.retrieval_type == "dense":
+            documents = []
+            for data, meta in data_collection:
+                if len(data) < 5:
+                    continue
+                metadata = {"source": meta}
+                new_doc = Document(page_content=data, metadata=metadata)
+                documents.append(new_doc)
+            assert documents!= [], "The given file/files cannot be loaded."
+            vectordb = Chroma.from_documents(documents=documents, embedding=self.embeddings,
+                                             persist_directory=self.persist_dir)
+            vectordb.persist()
+            print("The local knowledge base has been successfully built!")
+            return vectordb
+        elif self.retrieval_type == "sparse":
+            if self.document_store == "inmemory":
+                document_store = InMemoryDocumentStore(use_gpu=False, use_bm25=True)
+            elif self.document_store == "Elasticsearch":
+                document_store = ElasticsearchDocumentStore(host="localhost", index=self.index_name,
+                                                            port=9200, search_fields=["content", "title"])

```

Comment:
shall we check whether port 9200 is available?

Reply:
this part is transferring from haystack to langchain. This part would be removed in yuxiang's PR